pymeasure/instruments/keithley/keithley2600.py

```diff
@@ -0,0 +1,323 @@
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2020 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+import time
+import numpy as np
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import truncated_range, strict_discrete_set
+
+# Setup logging
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class Keithley2600(Instrument):
+    """Represents the Keithley 2600 series (channel A and B) SourceMeter"""
+
+    def __init__(self, adapter, **kwargs):
+        super(Keithley2600, self).__init__(
+            adapter,
+            "Keithley 2600 SourceMeter",
+            **kwargs
+        )
+        self.ChA = Channel(self, 'a')
+        self.ChB = Channel(self, 'b')
+
+    @property
+    def error(self):
+        """ Returns a tuple of an error code and message from a
+        single error. """
+        err = self.ask('print(errorqueue.next())')
+        err = err.split('\t')
+        # Keithley Instruments Inc. sometimes on startup
+        # if tab delimitated message is greater than one, grab first two as code, message
+        # otherwise, assign code & message to returned error
+        if len(err) > 1:
+            err = (float(err[0]), err[1])
```

Comment:
Error codes are really floats?

Reply:
As far as I know error codes are ints

Reply:
then you should not cast to float, but to int, no?

Reply:
That's what I would do, but since this function was written by @mcdo0486 if I'm not wrong, maybe I'd wait his/her response on this! :D

Reply:
Feel free to change the error() and check_error() code. Yea, it should be cast to int